monitoring/mock_uss/interaction_logging/logger.py

```diff
@@ -1,44 +1,34 @@
-import functools
 import os
 import datetime
-import copy
-from loguru import logger
-from typing import List, Optional
 
 import flask
 import json
-from typing import Dict
+from loguru import logger
 
-from monitoring.monitorlib.fetch import QueryError, Query, describe_flask_query
-from monitoring.monitorlib.mock_uss_interface.interaction_log import (
-    Interaction,
-    Issue,
-)
 from monitoring.mock_uss import webapp, require_config_value
 from monitoring.mock_uss.interaction_logging.config import KEY_INTERACTIONS_LOG_DIR
-from flask import Request, Response
+from monitoring.mock_uss.interaction_logging.interactions import (
+    Interaction,
+    QueryDirection,
+)
+from monitoring.monitorlib.clients import QueryHook, query_hooks
+from monitoring.monitorlib.fetch import Query, describe_flask_query, QueryType
 
 require_config_value(KEY_INTERACTIONS_LOG_DIR)
 
 
-def log_interaction(
-    direction: str, type: str, query: Query, issues: Optional[List[Issue]]
-):
-    """
-    Logs the REST calls between Mock USS to SUT
+def log_interaction(direction: QueryDirection, query: Query) -> None:
+    """Logs the REST calls between Mock USS to SUT
     Args:
-        direction: incoming or outgoing
-        method: GET or POST
-        type: Op, Pos or Constr
-    Returns:
-
+        direction: Whether this interaction was initiated or handled by this system.
+        query: Full description of the interaction to log.
     """
-    interaction: Interaction = Interaction(query=query, reported_issues=issues)
+    interaction: Interaction = Interaction(query=query, direction=direction)
     method = query.request.method
-    log_file(f"{direction}_{method}_{type}", interaction)
+    log_file(f"{direction}_{method}", interaction)
```

Comment:
We should add type also, to distinguish between op, constraint and telemetry type logs.

Reply:
Why is "type" in the filename useful?  Each object is loaded regardless of filename any way.  Also, "type" is not clearly defined as there are many other interaction types besides op, constraint, and telemetry.

Reply:
That's right. I think is more for readability while we are manually checking. As such then we don't need method and direction as well in the name.